plugin/pkg/scheduler/algorithm/well_known_labels.go

```diff
@@ -20,7 +20,7 @@ const (
 	// When feature-gate for TaintBasedEvictions=true flag is enabled,
 	// TaintNodeNotReady would be automatically added by node controller
 	// when node is not ready, and removed when node becomes ready.
-	TaintNodeNotReady = "node.alpha.kubernetes.io/notReady"
+	TaintNodeNotReady = "node.alpha.kubernetes.io/not-ready"
```

Comment:
If we're going to break compat, we should get rid of the .alpha. and update the docs.

Reply:
Should it be `"node.beta.kubernetes.io/not-ready"` or `"node.kubernetes.io/not-ready"`?

The doc PR is already proposed: https://github.com/kubernetes/kubernetes.github.io/pull/5188

Reply:
I thought we said no more alpha/beta anywhere and use feature gates instead. xref https://github.com/kubernetes/community/blob/master/contributors/devel/api_changes.md#alpha-beta-and-stable-versions which mentions not using an annotation for what should be a field. Not sure if there's official guidance on using alpha in the name of an annotation key. cc @liggitt

Reply:
> If we're going to break compat, we should get rid of the .alpha. and update the docs.

in general, I agree... every time this changes, it is disruptive for existing workloads. With an auto-added taint, it's tricky to know what we expect for compatible behavior:
* if we add the old and new taints, workloads have to tolerate both to be scheduled, which propagates use of the thing we're trying to deprecate
* if we only add the new one, old workloads that tolerated the old taint will fail

Reply:
At least it's an alpha feature that we're discussing. I'd vote for a hard break.

Reply:
@kubernetes/api-reviewers

Reply:
Agree with @ncdc @timothysc here, a hard break is inevitable. 

 This taint is not apply to feature gate since it's not a filed. I would vote for change it to `node.beta.kubernetes.io/not-ready` or even `"node.kubernetes.io/not-ready"` in next release.

@liggitt We can add a validation logic in node controller, since user must restart kubelet during upgrade. But I'm really not sure if it's worthy. WDYT?

Reply:
`node.kubernetes.io/not-ready` SGTM. Breaking alpha is fine for me.

Reply:
+1 to `node.kubernetes.io/not-ready`. Please don't include alpha/beta in the key name.

Reply:
Ok, so this is gonna be `node.kubernetes.io/not-ready` in v1.9 then when this moves to beta, right?

Reply:
I think so, I will rebase @gmarek 's commit and update this PR. As well docs and community repo.

Reply:
to be clear, what are we expecting to happen to existing nodes with the old alpha taint? will it be automatically removed?

Reply:
@liggitt I think the "hard break" means existing nodes with old taint will not work.

Reply:
@liggitt we presumably would need to write code that removes the old alpha taint, wouldn't we? Otherwise it would just stay there, ignored, albeit a bit confusing.

Reply:
> Otherwise it would just stay there, ignored, albeit a bit confusing.

a taint won't get ignored... it'll require every workload to be aware of and tolerate it

Reply:
Oh, right, so we need a plan to remove it. It would be easiest for the developers if we could just put that in the release notes (run X to remove/update this old taint key), but it would be easiest for users if they didn't have to do anything special...

Reply:
+1 for users having to not care about this stuff... or are the affected users here only ones that have opted into this feature?

Reply:
We can add a special case in taint-controller to ignore those alpha taints. We've done such things before.

Reply:
> We can add a special case in taint-controller to ignore those alpha taints. We've done such things before.

things like that make the behavior of the system extremely hard to understand. If we want to give a path for nodes with that taint, we should migrate it via the API so the behavior is observable and consistent

Reply:
@luxas I guess the affected users are limited, as we have a more well known `NoSchedule` taint to do similar thing.

If we decide not to rely on user to replace the taint (I agree user may ignore release note), I will vote for ignoring this taint @gmarek suggest. In this case users will know the old taint does not work, and we can also event this out.

@liggitt but I'm not quite clear about migrating it via API. Mind explanation?

Another approach in my mind is we c

Reply:
> I will vote for ignoring this taint as @gmarek suggest

special cases hidden internally are not inspectable. it can lead to bad outcomes where a user sees a node with the `node.alpha.kubernetes.io/notReady` taint and assumes (reasonably) that pods will not be scheduled there.

> @liggitt but I'm not quite clear about migrating it via API.

I mean that if we have processes that manage the `node.alpha.kubernetes.io/notReady` taint today, we could have those simply remove that taint at the same time they are setting/updating the `node.kubernetes.io/not-ready` taint. That way, the change is observable via the API.

Reply:
@liggitt The current processes (and we plan to) is add in release note and warn user. e.g. https://groups.google.com/forum/#!topic/kubernetes-dev/-fwfHQtm9GE, https://github.com/kubernetes/kubernetes/issues/25318

Personally, I think it's enough, since 1. we will break compatibility anyhow, 2. the impact scope is so big.

Reply:
We can also write a admission plugin that will do the substitution, but that'll work only for things we update.

Maybe we should have a mechanism that assures that all objects are touched after the upgrade?

Reply:
Can we do it in node controller?

Reply:
@gmarek I think the substitution in admission is enough as it seems taints only affect pods and nodes. Other objects in your mind?